Mathlib/Computability/PartrecCode.lean

```diff
@@ -1015,4 +1015,18 @@ theorem fixed_point₂ {f : Code → ℕ →. ℕ} (hf : Partrec₂ f) : ∃ c :
 
 end
 
+/-- There are only countably many partial recursive functions ℕ → ℕ. -/
+instance : Countable {f : ℕ →. ℕ // Nat.Partrec f} := by
+  choose v hv using fun x : {f : ℕ →. ℕ // Nat.Partrec f} ↦ Nat.Partrec.Code.exists_code.1 x.2
+  suffices v.Injective from this.countable
+  intro f g h
+  ext
+  simp [h, ← hv]
+
+/-- There are only countably many computable functions ℕ → ℕ. -/
+instance : Countable {f : ℕ → ℕ // Computable f} :=
+  @Function.Injective.countable {f : ℕ → ℕ // Computable f} {f : ℕ →. ℕ // Nat.Partrec f} _
+    (fun f => ⟨f.val, Partrec.nat_iff.mp f.2⟩)
+    fun _ _ h => SetCoe.ext <| PFun.lift_injective <| (Subtype.mk.injEq _ _ _ _) ▸ h
```

Comment:
```suggestion
/-- There are only countably many partial recursive partial functions `ℕ →. ℕ`. -/
instance : Countable {f : ℕ →. ℕ // _root_.Partrec f} := by
  apply Function.Surjective.countable (f := fun c => ⟨eval c, eval_part.comp (.const c) .id⟩) 
  intro ⟨f, hf⟩; simpa using exists_code.1 hf

/-- There are only countably many computable functions `ℕ → ℕ`. -/
instance : Countable {f : ℕ → ℕ // Computable f} :=
  @Function.Injective.countable {f : ℕ → ℕ // Computable f} {f : ℕ →. ℕ // _root_.Partrec f} _
    (fun f => ⟨f.val, f.2⟩)
    (fun _ _ h => Subtype.val_inj.1 (PFun.lift_injective (by simpa using h)))
```

Reply:
I think the second proof would be clearer if it was in tactic mode. Something like this:
```lean 4
instance : Countable {f : ℕ → ℕ // Computable f} := by
  let g : {f : ℕ → ℕ // Computable f} → {f : ℕ →. ℕ // _root_.Partrec f} := fun f => ⟨f.1, f.2⟩
  refine Function.Injective.countable (f := g) fun _ _ h => Subtype.val_inj.1 (PFun.lift_injective ?_)
  simpa [g] using h
```
I'd probably split the `refine` block further, but I think that's up to taste.

Reply:
I guess there are different philosophies --- (1) keep it succinct since mathematically it's not an interesting proof, or (2) make it easy to understand even if it's longer. Also what's clearer may depend on the reader and their taste/experience. :)

Reply:
I do want to mention that having more readable proofs can still be useful for "trivial" results, especially if they have to be refactored later down the line. Bespoke term mode proofs have caused me trouble in the ordinal part of the library in the past...

Reply:
Makes sense... well I'm fine with anything